### PR TITLE
feat: Adding calls to custom style hooks derived from context in react-dialog components

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Changes
- Updated components to call custom style hook

## Issues
- Updates #27058

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/efb2274e37c0237f3a10a4630b2696a9880f6398'>efb2274e37c0237f3a10a4630b2696a9880f6398</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Avatar-mount_base.html">1669</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Avatar-mount.html">1700</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Button-mount_base.html">1125</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Button-mount.html">1102</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Field-mount_base.html">2435</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Field-mount.html">2673</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProvider-mount_base.html">1953</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProvider-mount.html">2058</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-mount_base.html">758</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-mount.html">761</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">725</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">715</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">736</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">760</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/InfoButton-mount_base.html">672</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/InfoButton-mount.html">681</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/MakeStyles-mount_base.html">2298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/MakeStyles-mount.html">2279</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Persona-mount_base.html">3748</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/Persona-mount.html">3690</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/SpinButton-mount_base.html">3044</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27081/perf-test-react-components/SpinButton-mount.html">2957</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-dialog-396e0302-47d0-4e6f-84ab-c29537896234.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-dialog-396e0302-47d0-4e6f-84ab-c29537896234.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 44 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27081&commitId=423130db7b24f969b301efe584d91a14b4125b46&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1824 screenshots unchanged, 0 screenshots with different dimensions and 44 screenshots with visible difference.


  <details><summary><b>unknown</b> 44 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Dialog.actions position start & position end - Dark Mode.chromium.png|810616| Changed|
|Dialog.actions position start & position end - High Contrast.chromium.png|833866| Changed|
|Dialog.actions position start & position end - RTL.chromium.png|913411| Changed|
|Dialog.actions position start & position end.chromium.png|913412| Changed|
|Dialog.actions position start - Dark Mode.chromium.png|810257| Changed|
|Dialog.actions position start - High Contrast.chromium.png|833479| Changed|
|Dialog.actions position start - RTL.chromium.png|913743| Changed|
|Dialog.actions position start.chromium.png|913744| Changed|
|Dialog.alert - Dark Mode.chromium.png|810257| Changed|
|Dialog.alert - High Contrast.chromium.png|833477| Changed|
|Dialog.alert - RTL.chromium.png|913743| Changed|
|Dialog.alert.chromium.png|913744| Changed|
|Dialog.default - Dark Mode.chromium.png|810257| Changed|
|Dialog.default - High Contrast.chromium.png|833477| Changed|
|Dialog.default - RTL.chromium.png|913743| Changed|
|Dialog.default.chromium.png|913744| Changed|
|Dialog.nested - Dark Mode.chromium.png|813312| Changed|
|Dialog.nested - High Contrast.chromium.png|835010| Changed|
|Dialog.nested - RTL.chromium.png|908834| Changed|
|Dialog.nested.chromium.png|908834| Changed|
|Dialog.no actions - Dark Mode.chromium.png|818158| Changed|
|Dialog.no actions - High Contrast.chromium.png|837204| Changed|
|Dialog.no actions - RTL.chromium.png|908568| Changed|
|Dialog.no actions.chromium.png|908568| Changed|
|Dialog.no title & no actions - Dark Mode.chromium.png|825529| Changed|
|Dialog.no title & no actions - High Contrast.chromium.png|842452| Changed|
|Dialog.no title & no actions - RTL.chromium.png|902475| Changed|
|Dialog.no title & no actions.chromium.png|902476| Changed|
|Dialog.no title - Dark Mode.chromium.png|817628| Changed|
|Dialog.no title - High Contrast.chromium.png|838726| Changed|
|Dialog.no title - RTL.chromium.png|907650| Changed|
|Dialog.no title.chromium.png|907651| Changed|
|Dialog.non-modal - Dark Mode.chromium.png|1466253| Changed|
|Dialog.non-modal - High Contrast.chromium.png|1491070| Changed|
|Dialog.non-modal - RTL.chromium.png|1597111| Changed|
|Dialog.non-modal.chromium.png|1597113| Changed|
|Dialog.scroll long content - Dark Mode.chromium.png|665839| Changed|
|Dialog.scroll long content - High Contrast.chromium.png|704720| Changed|
|Dialog.scroll long content - RTL.chromium.png|1052799| Changed|
|Dialog.scroll long content.chromium.png|1052823| Changed|
|Dialog.title custom action - Dark Mode.chromium.png|817381| Changed|
|Dialog.title custom action - High Contrast.chromium.png|836611| Changed|
|Dialog.title custom action - RTL.chromium.png|909167| Changed|
|Dialog.title custom action.chromium.png|909167| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: change/@fluentui-react-dialog-396e0302-47d0-4e6f-84ab-c29537896234.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "type": "minor",
+  "comment": "feat: custom styles",
```
Comment (khmakoto) [inline change/@fluentui-react-dialog-396e0302-47d0-4e6f-84ab-c29537896234.json:3]:
```suggestion
  "comment": "feat: Adding calls to custom style hooks derived from context.",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**